flocker/control/httpapi.py

```diff
@@ -90,9 +92,44 @@
     code=CONFLICT, description=u"Lease already held.")
 NODE_BY_ERA_NOT_FOUND = make_bad_request(
     code=NOT_FOUND, description=u"No node found with given era.")
+
 _UNDEFINED_MAXIMUM_SIZE = object()
 
 
+def get_configuration_tag(api):
+    """
+    Return tag value for the configuration.
+
+    :param ConfigurationAPIUserV1 api: API instance.
+    :return: Tag as ``bytes``.
+    """
+    return api.persistence_service.configuration_hash()
+
+
+def _if_configuration_matches(original):
+    """
+    Decorator that compares X-If-Configuration-Matches header to result of
+    ``get_configuration_tag``.
+
+    :param original: Original function.
+    :return: Wrapped function.
+    """
+    @wraps(original)
+    def render_if_matches(self, request, **route_arguments):
+        if request.requestHeaders.hasHeader(b"X-If-Configuration-Matches"):
+            tag = get_configuration_tag(self)
+            if_matches = request.requestHeaders.getRawHeaders(
+                b"X-If-Configuration-Matches")
+            if tag not in if_matches:
+                request.setResponseCode(PRECONDITION_FAILED)
```

Comment:
:question: Could we add the actual tag here too to save having to make a further request?